presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,451 @@
+        assertQuery("select ST_YMax(ST_GeometryFromText('LineString(8 4, 5 7)'))", "select 7.0");
+        assertQuery("select ST_XMax(ST_GeometryFromText('MultiLineString((1 1, 5 1), (2 4, 4 4))'))", "select 5");
+        assertQuery("select ST_YMax(ST_GeometryFromText('MultiLineString((1 1, 5 1), (2 4, 4 4))'))", "select 4");
+        assertQuery("select ST_XMax(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "select 3.0");
+        assertQuery("select ST_YMax(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "select 1.0");
+        assertQuery("select ST_XMax(ST_GeometryFromText('MultiPolygon (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))'))", "select 6.0");
+        assertQuery("select ST_YMax(ST_GeometryFromText('MultiPolygon (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 10, 6 4)))'))", "select 10.0");
+    }
+
+    @Test
+    public void testSTMin()
+            throws Exception
+    {
+        assertQuery("select ST_XMin(ST_GeometryFromText('Point (1.5 2.5)'))", "select 1.5");
+        assertQuery("select ST_YMin(ST_GeometryFromText('Point (1.5 2.5)'))", "select 2.5");
+        assertQuery("select ST_XMin(ST_GeometryFromText('MultiPoint (1 2, 2 4, 3 6, 4 8)'))", "select 1");
+        assertQuery("select ST_YMin(ST_GeometryFromText('MultiPoint (1 2, 2 4, 3 6, 4 8)'))", "select 2");
+        assertQuery("select ST_XMin(ST_GeometryFromText('LineString(8 4, 5 7)'))", "select 5.0");
+        assertQuery("select ST_YMin(ST_GeometryFromText('LineString(8 4, 5 7)'))", "select 4.0");
+        assertQuery("select ST_XMin(ST_GeometryFromText('MultiLineString((1 1, 5 1), (2 4, 4 4))'))", "select 1");
+        assertQuery("select ST_YMin(ST_GeometryFromText('MultiLineString((1 2, 5 3), (2 4, 4 4))'))", "select 2");
+        assertQuery("select ST_XMin(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "select 2.0");
+        assertQuery("select ST_YMin(ST_GeometryFromText('Polygon ((2 0, 2 1, 3 1))'))", "select 0.0");
+        assertQuery("select ST_XMin(ST_GeometryFromText('MultiPolygon (((1 10, 1 3, 3 3, 3 10)), ((2 4, 2 6, 6 6, 6 4)))'))", "select 1.0");
+        assertQuery("select ST_YMin(ST_GeometryFromText('MultiPolygon (((1 10, 1 3, 3 3, 3 10)), ((2 4, 2 6, 6 10, 6 4)))'))", "select 3.0");
+    }
+
+    @Test
+    public void testSTNumInteriorRing()
+            throws Exception
+    {
+        assertQuery("select ST_NumInteriorRing(ST_GeometryFromText('POLYGON ((0 0, 0 5, 5 5, 5 0, 0 0))'))", "select 0");
+        assertQuery("select ST_NumInteriorRing(ST_GeometryFromText('POLYGON ((0 0, 8 0, 0 8, 0 0), (1 1, 1 5, 5 1, 1 1))'))", "select 1");
+        assertQueryFails("select ST_NumInteriorRing(ST_GeometryFromText('LineString(8 4, 5 7)'))", "ST_NumInteriorRing only applies to Polygon. Input type is: LineString");
+        assertQueryFails("select ST_NumInteriorRing(ST_GeometryFromText('POLYGON EMPTY'))", "ST_NumInteriorRing only applies to Polygon. Input type is: MultiPolygon");
```

Comment:
According to the spec, ST_NumInteriorRing of an empty Polygon should return null.